File: sync/src/main/java/com/inceptionnotes/sync/Events.java
package com.inceptionnotes.sync;

import com.inceptionnotes.sync.events.Event;
import com.inceptionnotes.sync.events.IdentifyEvent;
import com.inceptionnotes.sync.events.SimpleMessageServerEvent;
import com.inceptionnotes.sync.events.SyncEvent;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by jacob on 1/26/18.
 */

public class Events {
    public static final Map<String, Class<? extends Event>> events = new HashMap<>();
    public static final Map<Class<? extends Event>, String> actions = new HashMap<>();

    static {
        events.put("sync", SyncEvent.class);
        events.put("identify", IdentifyEvent.class);
        events.put("message", SimpleMessageServerEvent.class);

        events.forEach((k, v) -> actions.put(v, k));
    }
}

File: sync/src/main/java/com/inceptionnotes/sync/util/Events.kt
package com.inceptionnotes.sync.util

import com.inceptionnotes.sync.events.*
import java.util.*

/**
 * Created by jacob on 1/26/18.
 */

object Events {
    val events: MutableMap<String, Class<out Event>> = HashMap()
    val actions: MutableMap<Class<out Event>, String> = HashMap()

    init {
        events["sync"] = SyncEvent::class.java
        events["identify"] = IdentifyEvent::class.java
        events["message"] = SimpleMessageServerEvent::class.java
        events["show"] = ShowEvent::class.java
        events["server"] = ServerEvent::class.java

        events.forEach { (k, v) -> actions[v] = k }
    }
}
